components/sponsor/class-sponsor.php

```diff
@@ -0,0 +1,274 @@
+	 *
+	 * @return array
+	 */
+	public static function get_fm_fields() {
+		return [
+			'label'   => new \Fieldmanager_Textfield(
+				[
+					'label'         => __( 'Eyebrow Label', 'civil-first-fleet' ),
+					'default_value' => self::get_default_eyebrow_label(),
+				]
+			),
+			'link'    => new \Fieldmanager_Link( __( 'Link', 'civil-first-fleet' ) ),
+			'logo_id' => new \Fieldmanager_Media( __( 'Logo', 'civil-first-fleet' ) ),
+			'message' => new \Fieldmanager_RichTextArea( __( 'Message', 'civil-first-fleet' ) ),
+		];
+	}
+
+	/**
+	 * Get the FM fields used to schedule sponsors.
+	 *
+	 * @return array
+	 */
+	public static function get_schedule_fm_fields() : array {
+		return [
+			'schedules' => new \Fieldmanager_Group(
+				[
+					'label'          => __( 'Sponsor Schedules', 'civil-first-fleet' ),
+					'limit'          => 0,
+					'extra_elements' => 0,
+					'label'          => __( 'New Schedule', 'civil-first-fleet' ),
+					'label_macro'    => [ '%s', 'sponsor_id' ],
+					'add_more_label' => __( 'Add Schedule', 'civil-first-fleet' ),
+					'collapsed'      => true,
+					'collapsible'    => true,
+					'sortable'       => true,
+					'children'       => [
+						'sponsor_id'      => new \Fieldmanager_Select(
+							[
+								'datasource' => new \Fieldmanager_Datasource_Post(
+									[
+										'query_args' => [
+											'post_type' => 'sponsor',
+										],
+									]
+								),
+							]
+						),
+						'enable_schedule' => new \Fieldmanager_Checkbox( __( 'Schedule this sponsor?', 'civil-first-fleet' ) ),
+						'start_date'      => new \Fieldmanager_Datepicker(
+							[
+								'label'      => __( 'Start Date', 'civil-first-fleet' ),
+								'display_if' => [
+									'src'   => 'enable_schedule',
+									'value' => true,
+								],
+							]
+						),
+						'end_date'        => new \Fieldmanager_Datepicker(
+							[
+								'label'      => __( 'End Date', 'civil-first-fleet' ),
+								'display_if' => [
+									'src'   => 'enable_schedule',
+									'value' => true,
+								],
+							]
+						),
+					],
+				]
+			),
+		];
+	}
+
+	/**
+	 * Parse the first valid sponsor from an array of scheduled sponsors.
+	 *
+	 * @param array $schedules Schedules.
+	 * @return self
+	 */
+	public function parse_from_schedule_fm_data( array $schedules ) : self {
+
+		// Loop through schedules to determine which sponsor should be displayed.
+		foreach ( $schedules as $schedule ) {
+
+			// Validate the schedule shape.
+			$schedule = wp_parse_args(
+				$schedule,
+				[
+					'enable_schedule' => false,
+					'end_date'        => '',
+					'sponsor_id'      => 0,
+					'start_date'      => '',
+				]
+			);
+
+			// If enable schedule is false, use this sponsor.
+			if ( ! filter_var( $schedule['enable_schedule'] ?? false, FILTER_VALIDATE_BOOLEAN ) ) {
+				$this->set_post( $schedule['sponsor_id'] ?? 0 );
+				return $this;
+			}
+
+			$current_time = time();
```

Comment:
Would using `current_time()` instead of `time()` work to address your timezone offsets @todo?